controllers/bounce_processes.go

```diff
@@ -50,28 +50,35 @@ func (b BounceProcesses) Reconcile(r *FoundationDBClusterReconciler, context ctx
 	minimumUptime := math.Inf(1)
 	addressMap := make(map[string]string, len(status.Cluster.Processes))
 	for _, process := range status.Cluster.Processes {
-		addressMap[process.Locality["instance_id"]] = process.Address
+		if _, ok := process.Locality["process_id"]; ok {
+			addressMap[process.Locality["process_id"]] = process.Address
+		} else {
+			addressMap[process.Locality["instance_id"]] = process.Address
+		}
+
 		if process.UptimeSeconds < minimumUptime {
 			minimumUptime = process.UptimeSeconds
 		}
 	}
 
 	addresses := make([]string, 0, len(cluster.Status.IncorrectProcesses))
 
-	for instanceID := range cluster.Status.IncorrectProcesses {
-
-		if addressMap[instanceID] == "" {
-			return false, fmt.Errorf("Could not find address for instance %s", instanceID)
+	for process := range cluster.Status.IncorrectProcesses {
+		if addressMap[process] == "" {
+			return false, fmt.Errorf("could not find address for process: %s", process)
 		}
 
-		addresses = append(addresses, addressMap[instanceID])
+		addresses = append(addresses, addressMap[process])
+
+		instanceID := GetInstanceIDFromProcessID(process)
+		log.Info("GetPod", "instanceID", instanceID)
```

Comment:
If we're going to leave this logging statement in, can we add some more context?

Reply:
In general, I think it's best to include the cluster name and namespace in all log messages that are specific to a cluster. This helps us narrow down relevant log messages when dealing with issues for a cluster.

Reply:
That makes sense, I remove these logging statements I just used them for debugging.